### PR TITLE
If a timeout is provided, don't use CURLOPT_LOW_SPEED_TIME and CURLOPT_LOW_SPEED_LIMIT

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This is for operations which may take a while but do not send a lot of data.  I was seeing timeouts in both CompleteMultipart and DeleteObject when using a local MinIO server.

The client using this library sets the timeout.

### CODE REVIEW

File: src/request.c [hunk 1]
```diff
@@ -1239,12 +1239,13 @@ static S3Status setup_curl(Request *request,
...
     // xxx todo - allow configurable max send and receive speed
-    curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, 1024);
-    curl_easy_setopt_safe(CURLOPT_LOW_SPEED_TIME, 15);
-
+    // jjames - only set the low speed limit and time if the timeout is 0
 
     if (params->timeoutMs > 0) {
         curl_easy_setopt_safe(CURLOPT_TIMEOUT_MS, params->timeoutMs);
+    } else {
+        curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, 256);
```
Comment (trel) [inline src/request.c]: why the change from 1024 above to 256 here?

otherwise, i think your comment is justification enough for moving it into the else block.

**Comment by trel** [comment]
do you want to reference issue 18 in the commit message?   if so, could make it a multiline commit message, so it doesn't wrap...

just considering a pull request upstream, later.

**Comment by JustinKyleJames** [comment]
> do you want to reference issue 18 in the commit message? if so, could make it a multiline commit message, so it doesn't wrap...
> 
> just considering a pull request upstream, later.

This is done.

**Comment by trel** [comment]
excellent - please # it.

**Comment by JustinKyleJames** [comment]
Done
________________________________
